src/main/kotlin/com/atlassian/performance/tools/jiraactions/page/IssueCreateDialog.kt

```diff
@@ -49,6 +51,37 @@ internal class IssueCreateDialog(
         return this
     }
 
+    /**
+     * Click 'Configure Fields' to display 'All' to ensure
+     * all mandatory fields are displayed in creation dialog.
+     *
+     */
+    fun showAllFields(): IssueCreateDialog {
+        // There might be some popup message dialogs, which may cover the 'Configure Fields' button
+        // Force dispose those popup dialogs
+        dismissAllPopups()
+        driver.wait(elementToBeClickable(configColumnField)).click()
+        val locator = By.xpath("//div[@id='inline-dialog-field_picker_popup']//a[text()='All']")
```

Comment:
This selector assumes the text will be "All". Will it work for other languages than English?

Reply:
No, but we can add alternative locators for other languages and `or` them.

Reply:
Is it the only way to select the field?

Reply:
Selecting by content is recommended by Jira Cloud. When the content is predictable, it's stable. When content is dynamic, like issue summary, they add an explicit attribute, like `data-test-id="issue.views.issue-base.foundation.summary.heading"`.
I think it's a good approach, because users look for content, rather than selectors.

Reply:
Users look for content, but the content is not an API. Explicit attributes (classes, ids) provide extension points so they may be treated as a public API (vendors rely on it). Text-based selectors are harder to maintain:
1. Text may change.
2. A new language may be added.

This PR introduces two text-based selectors ("please select" and "All"). 

Where can I read Jira Cloud's recommendation?

Reply:
> Text may change. A new language may be added.

True, but selectors may change as well. It's a matter of how frequently the "API" changes and how hard it is to fix when they do.

With content, it's trivial to learn how to extend it. You can see it on a screenshot. Not sure whether it changes more or less often than DOM.

> Where can I read Jira Cloud's recommendation?

I was told that in a conversation with an architect - Dariusz.

Reply:
Thanks for the comments, I will re-examine the dom to see if there is any better alternatives as selectors.

Reply:
I updated both xpath to not using text, I tested in HWR project manually and it worked. But I am getting some errors during setup when test the integration test:

org.openqa.selenium.WebDriverException: unknown error: Element is not clickable at point (1046, 232)
  (Session info: chrome=66.0.3359.117)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 4.9.184-linuxkit x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 0 milliseconds
Build info: version: '3.11.0', revision: 'e59cfb3', time: '2018-03-11T20:26:55.152Z'
System info: host: 'C02XV0AGJGH7', ip: 'fe80:0:0:0:cf7:988c:3da4:bc45%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.14.6', java.version: '1.8.0_191'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
Capabilities {acceptInsecureCerts: false, acceptSslCerts: false, applicationCacheEnabled: false, browserConnectionEnabled: false, browserName: chrome, chrome: {chromedriverVersion: 2.38.552522 (437e6fbedfa876..., userDataDir: /tmp/.org.chromium.Chromium...}, cssSelectorsEnabled: true, databaseEnabled: false, handlesAlerts: true, hasTouchScreen: false, javascriptEnabled: true, locationContextEnabled: true, mobileEmulationEnabled: false, nativeEvents: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, rotatable: false, setWindowRect: true, takesHeapSnapshot: true, takesScreenshot: true, unexpectedAlertBehaviour: , unhandledPromptBehavior: , version: 66.0.3359.117, webStorageEnabled: true, webdriver.remote.sessionid: 26c4f9dbb426da7b7e8144ecff8...}
Session ID: 26c4f9dbb426da7b7e8144ecff80a40d
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:214)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:166)
	at org.openqa.selenium.remote.http.JsonHttpResponseCodec.reconstructValue(JsonHttpResponseCodec.java:40)
	at org.openqa.selenium.remote.http.AbstractHttpResponseCodec.decode(AbstractHttpResponseCodec.java:80)
	at org.openqa.selenium.remote.http.AbstractHttpResponseCodec.decode(AbstractHttpResponseCodec.java:44)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
	at org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:279)
	at org.openqa.selenium.remote.RemoteWebElement.click(RemoteWebElement.java:83)
	at com.atlassian.performance.tools.dockerinfrastructure.jira.SetUpFromScratchAction.cleanErrorMessages(SetUpFromScratchAction.kt:112)
	at com.atlassian.performance.tools.dockerinfrastructure.jira.SetUpFromScratchAction.createSampleProject(SetUpFromScratchAction.kt:106)

Any thoughts?